rows.go

```diff
@@ -65,11 +66,19 @@ func (rows *mysqlRows) Columns() []string {
 }
 
 func (rows *mysqlRows) Close() (err error) {
+	if f := rows.finish; f != nil {
+		f()
+		rows.finish = nil
+	}
+
 	mc := rows.mc
 	if mc == nil {
 		return nil
 	}
-	if mc.netConn == nil {
+	if mc.isBroken() {
```

Comment:
ok. I just concerned about duplication of logic, because the main reasons of introducing function error was consolidation of this logic in one place. 
IMHO: code became simpler :)
``` go
if err := mc.error(); err != nil {
     return err
}
```